src/BloomExe/BookThumbNailer.cs

```diff
@@ -152,7 +151,34 @@ public Image GetThumbnailForPage(Book.Book book, IPage page, bool isLandscape)
 		/// <param name="thumbnailOptions"></param>
 		/// <param name="callback"></param>
 		/// <param name="errorCallback"></param>
-		public void RebuildThumbNail(Book.Book book, HtmlThumbNailer.ThumbnailOptions thumbnailOptions,
+		public void RebuildThumbNailAsync(Book.Book book, HtmlThumbNailer.ThumbnailOptions thumbnailOptions,
+			Action<BookInfo, Image> callback, Action<BookInfo, Exception> errorCallback)
+		{
+			RebuildThumbNail(book, thumbnailOptions, callback, errorCallback, true);
+		}
+
+		/// <summary>
+		/// Will call either 'callback' or 'errorCallback' UNLESS the thumbnail is readonly, in which case it will do neither.
+		/// </summary>
+		/// <param name="book"></param>
+		/// <param name="thumbnailOptions"></param>
+		/// <param name="callback"></param>
```

Comment:
Can you add a comment saying what the callback is for? I wouldn't expect this on a synchronous method.